### PR TITLE
src,fix: drop support for v0.2 and clean up contenttype

### PR DESCRIPTION
**Pull request by lance**
This commit removes support for the v0.2 specification. It also removes the
`contenttype` attribute from the `CloudEvent` object. While the HTTP protocol
binding specifies that in binary mode, the `datacontenttype` attribute should
map to the HTTP Content-Type header, that doesn't mean that the `CloudEvent`
object should have a `contenttype` property.

Fixes: https://github.com/cloudevents/sdk-javascript/issues/61
Fixes: https://github.com/cloudevents/sdk-javascript/issues/66

Signed-off-by: Lance Ball <lball@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,21 +91,21 @@ These are the supported specifications by this version.
 
 | **Specifications**                    | v0.1 | v0.2 | v0.3 | **v1.0** |
```
Comment (grant) [inline README.md:92]: We should just remove the column for v0.2. We already state we don't support 0.2 above.

**Comment by fabiojose** [comment]
@lance fix the conflicts, please.

**Comment by lance** [comment]
@fabiojose conflicts resolved

**Comment by fabiojose** [comment]
Do we already have the semantic version? Is necessary to fill out the changelog?

**Comment by fabiojose** [comment]
@lance something goes wrong with CI, can you check?

**Comment by grant** [comment]
Here's the error:

```
> cloudevents-sdk@1.0.0 coverage-publish /home/travis/build/cloudevents/sdk-javascript
> wget -qO - https://coverage.codacy.com/get.sh | sh -s report -l JavaScript -r coverage/lcov.info
sh: 3: set: Illegal option -o pipefail
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! cloudevents-sdk@1.0.0 coverage-publish: `wget -qO - https://coverage.codacy.com/get.sh | sh -s report -l JavaScript -r coverage/lcov.info`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the cloudevents-sdk@1.0.0 coverage-publish script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2020-04-28T20_38_50_402Z-debug.log
The command "if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then npm run coverage-publish; else npm test; fi" exited with 2.
cache.2
store build cache
Done. Your build exited with 1.
```

Something with the `coverage-publish` script.

**Comment by lance** [comment]
@fabiojose @grant This error is confusing. The coverage task didn't change with this commit. In fact, package.json didn't change at all. 

Hidden in the error message you pasted above is the real culprit.

> sh: 3: set: Illegal option -o pipefail

Here are the first three lines of the codacy https://coverage.codacy.com/get.sh script that is run in order to publish the code coverage stats.

```console
   1   │ #!/usr/bin/env bash
   2   │ 
   3   │ set -e +o pipefail
```

The error indicates that the get.sh script provided an illegal option to the `set` command on line 3. Again... seems weird. The `.travis.yaml` configuration file hasn't changed, so I don't know why all of a sudden the bash `set` command would begin failing. A simple test with a file like this indicates this should not fail.

```console
   1   │ #!/bin/bash
   2   │ 
   3   │ set -e +o pipefail
   4   │ 
   5   │ echo "hello!"
```

Maybe the build was run on a corrupted container? I don't have permissions to re-run the build on Travis-CI. But can you try that first?

**Comment by fabiojose** [comment]
Things went wrong by this PR #49, could you check @lance ?

**Pull request merged**